lib/migration.c

```diff
@@ -534,6 +534,60 @@ migration_region_access(vfu_ctx_t *vfu_ctx, char *buf, size_t count,
     return count;
 }
 
+bool
+migration_feature_supported(uint32_t flags) {
+    switch (flags & VFIO_DEVICE_FEATURE_MASK) {
+        case VFIO_DEVICE_FEATURE_MIGRATION:
+            return !(flags & VFIO_DEVICE_FEATURE_SET); // not supported for set
+        case VFIO_DEVICE_FEATURE_MIG_DEVICE_STATE:
+            return true;
+        default:
+            return false;
+    };
+}
+
+ssize_t
+migration_feature_get(vfu_ctx_t *vfu_ctx, uint32_t flags, void *buf)
+{
+    struct vfio_user_device_feature_migration *res;
+    struct vfio_user_device_feature_mig_state *state;
+
+    switch (flags & VFIO_DEVICE_FEATURE_MASK) {
+        case VFIO_DEVICE_FEATURE_MIGRATION:
+            res = buf;
+            // FIXME are these always supported? Can we consider to be
+            //   "supported" if said support is just an empty callback?
+            res->flags = VFIO_MIGRATION_STOP_COPY | VFIO_MIGRATION_PRE_COPY;
```

Comment:
Thanos?

Reply:
We can drop `VFIO_MIGRATION_PRE_COPY` for now. @w-henderson have you checked the kernel whether we need to return `RUNNING ` and `ERROR` as they're always supported? If we don't have to return them, we should have a comment.

Reply:
We don't need to return `RUNNING` or `ERROR` because they're always supported... interestingly we do need to return `STOP_COPY` despite that also always being supported! From the kernel:

> If this `GET` succeeds, the `RUNNING` and `ERROR` states are always supported. Support for additional states is indicated via the flags field; at least `VFIO_MIGRATION_STOP_COPY` must be set.

I'll add a comment.